src/core/tests/tensor.cpp

```diff
@@ -88,5 +94,251 @@ TEST(tensor, get_byte_size_u6_not_even_div_by_storage_unit) {
     const auto tensor = Tensor(element::u6, Shape{17});
     EXPECT_EQ(tensor.get_byte_size(), 3 + 4 * 3);
 }
+
+template <typename TypeParam>
+class ParametredOffloadTensorTest : public ::testing::Test {
+public:
+    using path_type = typename std::tuple_element<0, TypeParam>::type;
+    using element_type = typename std::tuple_element<1, TypeParam>::type;
+    static constexpr ov::element::Type ov_type = ov::element::from<element_type>();
+
+    void SetUp() override {
+        shape = {10, 20, 30, 40};
+        auto static_shape = shape.get_shape();
+        initial_tensor = Tensor(ov_type, static_shape);
+        std::vector<float> init_values(initial_tensor.get_size());
+        ov::test::utils::fill_data_random(init_values.data(), initial_tensor.get_size(), 10, 0, 100);
+
+        std::transform(init_values.begin(),
+                       init_values.end(),
+                       element::iterator<ov_type>(initial_tensor.data()),
+                       [](float value) {
+                           return static_cast<element_type>(value);
+                       });
+
+        file_name_str = ov::test::utils::generateTestFilePrefix();
+        file_name = path_type(file_name_str.begin(), file_name_str.end());
+    }
+
+    void remove_file() {
+        if (std::filesystem::exists(file_name))
```

Comment:
Use OV util functions this function can throw.

Reply:
it's just a test :)